### PR TITLE
RUN-4331 - Multiple channels by channelName

### PR DESCRIPTION
**Pull request by tgoc99**
Ability to create multiple channels for a single application.  Must use a channelName when you create a channel and you connect to channel by a channelName as well.  ChannelNames will be unique and creation will fail if you provide a name that is not unique.

[Core PR](https://github.com/HadoukenIO/core/pull/524)

Updated all channel related tests, plus there are six new testrunner tests and the additional js-adapter test in the code change:
[one](https://testing-dashboard.openfin.co/#/app/tests/5b7599ab6107b45e6d1eba2b/edit)
[two](https://testing-dashboard.openfin.co/#/app/tests/5b759e5a6107b45e6d1eba32/edit)
[three](https://testing-dashboard.openfin.co/#/app/tests/5b75d86c6107b45e6d1eba3e/edit)
[four](https://testing-dashboard.openfin.co/#/app/tests/5b75d9e56107b45e6d1eba3f/edit)
[five](https://testing-dashboard.openfin.co/#/app/tests/5b7ecb7c6107b45e6d1ebb45/edit)
[six](https://testing-dashboard.openfin.co/#/app/tests/5b7eca916107b45e6d1ebb44/edit)

[Win7](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5b8056906107b45e6d1ebb6b)
[Win10](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5b804ed76107b45e6d1ebb68)

### CODE REVIEW

File: src/api/interappbus/channel/index.ts [hunk 1]
```diff
@@ -36,67 +37,65 @@ export class Channel extends EmitterBase {
...
     }
 
-    // DOCS - if want to send payload, put payload in options
-    public async connect(options: Options): Promise<ChannelClient> {
-        const { uuid, name } = options;
-        let resolver: any;
-        let listener: any;
-        //@ts-ignore
-        // tslint:disable-next-line
+    public async connect(channelName: string, options?: Options): Promise<ChannelClient> {
```
Comment (pbaize) [inline src/api/interappbus/channel/index.ts]: If options is optional, repeating channelName shouldn't be on the interface. Also is uuid still necessary?

**Comment by datamadic** [comment]
@tgoc99 Are the corresponding doc updates going to be captured [here](https://appoji.jira.com/browse/RUN-4445)?

**Comment by tgoc99** [comment]
@datamadic - the javascript-adapter docs PR is [here](https://github.com/openfin/javascript-adapter/pull/482).  They do not yet include tutorials.  I think we could likely copy that docs file straight over from the javascript-adapter but haven't tried it yet.  Don't want the docs in two places yet as changes are still being made to the channel API.  I added js-adapter docs to that JIRA that you referenced, looking like it will be early next sprint or during cool-down.

File: src/api/base.ts [hunk 2]
```diff
@@ -90,7 +90,8 @@ export class EmitterBase extends Base {
             const newRefCount = refCount - 1;
             if (newRefCount === 0) {
                 await this.wire.sendAction('unsubscribe-to-desktop-event', runtimeEvent);
-                if (emitter.eventNames().length === 0) {
+                // TO DO: FIX - This was erroring out saying eventNames is not a function.
```
[the remark this answers is not in the document]
Comment (tgoc99) [reply src/api/base.ts]: removed comment

**Pull request merged**